hs-bindgen/src/HsBindgen/SHs/AST.hs

```diff
@@ -150,9 +150,9 @@ data SExpr ctx =
   deriving stock (Show)
 
 pattern EInt :: Int -> SExpr be
-pattern EInt i <- EIntegral (fromInteger -> i) (Just HsPrimCInt)
+pattern EInt i <- EIntegral (fromInteger -> i) (Just HsPrimInt)
   where
-    EInt i = EIntegral (fromIntegral i) (Just HsPrimCInt)
```

Comment:
Do we have cases with `Nothing` in `EIntegral`? We do, the enum values for example. We *could* be explicit there as well if we really want to; thus being always explicit about types of integral literals.